dds/DCPS/ReactorTask.cpp

```diff
@@ -64,51 +77,42 @@ int ReactorTask::open_reactor_task(void*, TimeDuration timeout,
   // Set our reactor and proactor pointers to a new reactor/proactor objects.
   if (use_async_send_) {
 #if defined (ACE_WIN32) && defined (ACE_HAS_WIN32_OVERLAPPED_IO)
-    reactor_ = new ACE_Reactor(new ACE_WFMO_Reactor, 1);
+    if (!reactor_) {
+      reactor_ = new ACE_Reactor(new ACE_WFMO_Reactor, 1);
 
-    ACE_WIN32_Proactor* proactor_impl = new ACE_WIN32_Proactor(0, 1);
-    proactor_ = new ACE_Proactor(proactor_impl, 1);
-    reactor_->register_handler(proactor_impl, proactor_impl->get_handle());
+      ACE_WIN32_Proactor* proactor_impl = new ACE_WIN32_Proactor(0, 1);
+      proactor_ = new ACE_Proactor(proactor_impl, 1);
+      reactor_->register_handler(proactor_impl, proactor_impl->get_handle());
+    }
 #else
-    reactor_ = new ACE_Reactor(new ACE_Select_Reactor, true);
-    proactor_ = 0;
+    if (!reactor_) {
+      reactor_ = new ACE_Reactor(new ACE_Select_Reactor, true);
+      proactor_ = 0;
+    }
 #endif
   } else {
-    reactor_ = new ACE_Reactor(new ACE_Select_Reactor, true);
-    proactor_ = 0;
+    if (!reactor_) {
+      reactor_ = new ACE_Reactor(new ACE_Select_Reactor, true);
+      proactor_ = 0;
+    }
   }
```

Comment:
It's bothered me for awhile that the else blocks are the same here. Maybe something like this?

```c++
#if defined (ACE_WIN32) && defined (ACE_HAS_WIN32_OVERLAPPED_IO)
  if (use_async_send_ && !reactor_) {
    reactor_ = new ACE_Reactor(new ACE_WFMO_Reactor, 1);

    ACE_WIN32_Proactor* proactor_impl = new ACE_WIN32_Proactor(0, 1);
    proactor_ = new ACE_Proactor(proactor_impl, 1);
    reactor_->register_handler(proactor_impl, proactor_impl->get_handle());
  } else
#endif
  if (!reactor_) {
    reactor_ = new ACE_Reactor(new ACE_Select_Reactor, true);
    proactor_ = 0;
  }
```